Source/TrieRouteMatcher.swift

```diff
@@ -79,8 +85,7 @@ public struct TrieRouteMatcher: RouteMatcherType {
             return nil
         }
 
-        let components = path.split("/")
-
+        let components = [request.method.description] + path.split("/")
```

Comment:
We don't have the same issue here because the components are already split by `/` (meaning its just an array of strings, such as `["GET","my","path"]` which is exactly what we want.